packages/block-editor/src/components/url-input/index.js

```diff
@@ -134,7 +134,20 @@ class URLInput extends Component {
 			return;
 		}
 
-		const isInitialSuggestions = ! ( value && value.length );
+		// Test for whitespace only:
+		// 1. must have a length - otherwise test from #1 will be truthy for string with no characters.
+		// 2. if all whitespace is stripped must be empty.
+		const containsOnlyWhitespace = value?.length && value.trim() === '';
+
+		// Initial suggestions are those shown when no search has been made.
+		// The criteria for "no search" are:
+		// 1. No value.
+		// 2. Value must not be entirely whitespace.
+		const isInitialSuggestions =
+			! value?.length && ! containsOnlyWhitespace;
+
```

Comment:
@getdave Continuing the discussion from the previous comment – my point is that `containsOnlyWhitespace` doesn't do anything right now. We get a tautology:

```js
const containsOnlyWhitespace = value?.length && value.trim() === '';
const isInitialSuggestions = ! value?.length && ! containsOnlyWhitespace;
```

Let's inline `containsOnlyWhitespace`:

```js
const isInitialSuggestions = ! value?.length && ! ( value?.length && value.trim() === '' );
```

Let's expand:

```js
const isInitialSuggestions = ! value?.length && ! value?.length || ! value?.length && ! value.trim() === '';
```

Let's simplify:

```js
const isInitialSuggestions = ! value?.length && ! value.trim() === '';
```

Trimmed value will already be empty when the length is 0, so we finally get: 

```js
const isInitialSuggestions = ! value?.length;
```

Which is what we have now.

So here's what I propose instead:

```suggestion
		const isInitialSuggestions = value?.length;

```

Of course having a comment with an explanation would still be useful.

Reply:
Ah I see. I've just inverted your suggestion (you put `value?.length;` instead of `!value?.length;` and now it works.

Thanks for simplifying. I vastly overcomplicated this.

Reply:
Aw man, it's like [Muphry's law](https://en.wikipedia.org/wiki/Muphry%27s_law) for code 🙈 

> If you write anything criticizing editing or proofreading, there will be a fault of some kind in what you have written

Reply:
Thanks for the contribution. It greatly simplified this over complex code.